src/firecracker/src/main.rs

```diff
@@ -397,6 +422,51 @@ fn main_exec() -> Result<(), MainError> {
     }
 }
 
+/// Attempts to resize the processes file descriptor table to match RLIMIT_NOFILE or 2048, whichever
```

Comment:
Where does 2048 come from?

Reply:
Since we no longer are in the jailer, we cannot assume that a limit is actually set. 2048 is the default from the jailer

Reply:
sure but we moved the change in Firecracker to avoid the assumption that we are spawned from the jailer. That makes me think that the 2048 thing is arbitrary and I wonder if there's a better limit that we should be using (for example `rlim_cur`.

Reply:
Mh, so on my laptop I have `limit.rlim_max = 1048576`. I don't think we want to allocate that large of an fd table. Although maybe if we document it and just tell the users to set rlimits correctly when running firecracker? Not sure which solution I prefer here tbh

Reply:
Synced offline, since using rlim_max would result in a very significant memory overhead, we decided to leave it at the arbitrary default of 2048 (which matches the default the jailer uses for RLIMIT_NOFILE)